src/backend/WorkflowManagement/WorkflowManager.cs

```diff
@@ -76,10 +77,10 @@ public static WorkflowQueueMessage GetFirstStage(string submissionId)
         /// Executes the action for the current stage in the workflow. After the action completes,
         /// returns a message for the next stage in the queue. 
         /// </summary>
-        public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg)
+        public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg, CancellationToken cancelToken)
         {
             //Execute the action
-            WorkflowStage nextStage = await actions[(int)currentMsg.Stage].ExecuteAsync(currentMsg.SubmissionId);
+            WorkflowStage nextStage = await actions[(int)currentMsg.Stage].ExecuteAsync(currentMsg.SubmissionId, cancelToken);
```

Comment:
Since this could is a library, I would add `.ConfigureAwait(false)` if resuming on the same context does not matter.
```cs
WorkflowStage nextStage = await actions[(int)currentMsg.Stage].ExecuteAsync(currentMsg.SubmissionId, cancelToken).ConfigureAwait(false); 
```

Reply:
Sorry I missed it...done.